HelloWorldFunction/src/main/java/helloworld/App.java

```diff
@@ -22,102 +22,109 @@
         Map<String, String> headers = new HashMap<>();
         headers.put("Content-Type", "application/json");
         headers.put("X-Custom-Header", "application/json");
         try {
-            final String pageContents = this.getPageContents("https://checkip.amazonaws.com");
+            String pageContents = this.getPageContents("https://checkip.amazonaws.com");
             UUID id = UUID.randomUUID();
-            final String ddbtestput = this.PutItem("HelloWorldTable", "id", id.toString(), "field1", "stringValue1");
-            final String ddbtestget = this.GetItem("HelloWorldTable", "id", id.toString() );
+            String ddbtestput = this.PutItem("HelloWorldTable", "id", id.toString(), "field1", "stringValue1");
+            String ddbtestget = this.GetItem("HelloWorldTable", "id", id.toString());
 
-            String output = String.format("{ \"message\": \"hello world\", \"location\": \"%s\" \n \"put\":\"%s\"-\"%s\" \n \"get\":\"%s\"-\"%s\"}", pageContents,id.toString(),ddbtestput,id.toString(),ddbtestget);
+            String output = String.format(
+                    "{ \"message\": \"hello world\", \"location\": \"%s\" \n \"put\":\"%s\"-\"%s\" \n \"get\":\"%s\"-\"%s\"}",
+                    pageContents, id.toString(), ddbtestput, id.toString(), ddbtestget);
             return new GatewayResponse(output, headers, 200);
         } catch (IOException e) {
             return new GatewayResponse("{}", headers, 500);
         }
     }
 
-    private String getPageContents(String address) throws IOException{
+    private String getPageContents(String address) throws IOException {
         URL url = new URL(address);
-        try(BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()))) {
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()))) {
             return br.lines().collect(Collectors.joining(System.lineSeparator()));
         }
     }
 
-    private String GetItem(String TableName, String PrimaryKey, String Value) //throws AmazonServiceException
+    private String GetItem(String TableName, String PrimaryKey, String Value) // throws
+                                                                                                // AmazonServiceException
     {
 
-        HashMap<String,AttributeValue> key_to_get =
-        new HashMap<String,AttributeValue>();
-    
+        HashMap<String, AttributeValue> key_to_get = new HashMap<String, AttributeValue>();
+
         key_to_get.put(PrimaryKey, new AttributeValue(Value));
-        
+
         GetItemRequest request = null;
 
-        request = new GetItemRequest()
-            .withKey(key_to_get)
-            .withTableName(TableName);
-        
-        final AmazonDynamoDB ddb = AmazonDynamoDBClientBuilder.defaultClient();
-        
+        request = new GetItemRequest().withKey(key_to_get).withTableName(TableName);
+
+        AmazonDynamoDB ddb = AmazonDynamoDBClientBuilder.defaultClient();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is written so that the client cannot be reused across invocations of the Lambda function.
To improve the performance of the Lambda function, consider using static initialization/constructor, global/static variables and singletons. It allows to keep alive and reuse HTTP connections that were established during a previous invocation.
Learn more about [best practices for working with AWS Lambda functions](https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html).